### DISCUSSION TITLE
Apple privacyInfo manifest file - Questions and doubts?

### DISCUSSION THREAD

**Discussion by gabsamples6** [dotnet/maui#21462] [General]
Hi 
Like you I am going through the pain of creating a  PrivacyInfo.xcprivacy and filling all the data and wondering if someone could clarify the following points 

 NSPrivacyTracking 
- if you just store the userId on database for app functionality is this considered tracking?

NSPrivacyCollectedDataTypes
- if a datatype is not used I.E health and fitness,PreciseLocation etc..   do we still need to declare it as not used?
- strings provided by apple must be used for NSPrivacyCollectedDataTypePurposes - is this mandatory?
  it says "_Use the following strings as values for the NSPrivacyCollectedDataTypePurposes key in your NSPrivacyCollectedDataTypes dictionaries_:"
- if have a spelling mistake will it fail?
- many nugets used in xamarin are no longer maintained but we still use  what do you do? Most likely they will not bother? 
- how do you validate it? Surely you do not want to wait till the app is submitted, just to be told it failed!

would below represent a sample on how to fill a typical datatype 

     <dict>
         <key>NSPrivacyCollectedDataType</key>
         <string>NSPrivacyCollectedDataTypeEmailAddress</string>
         <key>NSPrivacyCollectedDataTypeLinked</key>
         <true/>
         <key>NSPrivacyCollectedDataTypeTracking</key>
         <false/>
         <key>NSPrivacyCollectedDataTypePurposes</key>
         <array>
             <string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
             <string>NSPrivacyCollectedDataTypePurposeProductPersonalization</string>
         </array>
     </dict>

I would appreciate any input on this..  

thanks

**Comment by davidortinau** [comment]
@jfversluis @mcumming do you have advice on these questions?

Our documentation doesn't mention those props specifically: https://learn.microsoft.com/en-us/dotnet/maui/ios/privacy-manifest?view=net-maui-8.0

**Reply by gabsamples6** [reply]
**UPDATE** 
I have spent time reading over and over the apple docs and what helped me pratically was to search in github for existing privacyinfo and what people have done and that gave me a pretty good idea of what to do and I have settled for the above as an example of data type

Having seen many privacy info files my conclusions (hopefully correct) are :

**Categories**
there are 4 categories
Even if not used I have noticed that people marked as false or empty array . Below is just an example...
```
  <key>NSPrivacyTracking</key>
  <false/>
  <key>NSPrivacyTrackingDomains</key>
  <array/>

```
**DataTypes**
Only declare what you use , and if you dont use do not mention it


**NSPrivacyAccessedAPITypes**

- I have included the Maui ones,  the rest I am struggling to be honest, to see I need to declare any
Xamarin - lost cause , who knows what some nugets do internally , and 100% owners will not bother..

**Still outstanding**
Apart from submitting is there any way we can validate this thing?  That would be too late in the game...

I hope this helps others. .
